Library/Homebrew/diagnostic.rb

```diff
@@ -748,14 +748,10 @@ def check_for_unreadable_installed_formula
 
       def check_for_unlinked_but_not_keg_only
         unlinked = Formula.racks.reject do |rack|
-          if (HOMEBREW_LINKED_KEGS/rack.basename).directory?
-            true
-          else
-            begin
-              Formulary.from_rack(rack).keg_only?
-            rescue FormulaUnavailableError, TapFormulaAmbiguityError
-              false
-            end
+          (HOMEBREW_LINKED_KEGS/rack.basename).directory? || begin
```

Comment:
Most of these autocorrect fixes are a bit gnarly.

Perhaps worth going through and doing the mixups manually rather than relying on autocorrect. For example this one will probably look better as

```suggestion
          next true if (HOMEBREW_LINKED_KEGS/rack.basename).directory?
          
          begin
```

Reply:
Yeah these did not look right to me either, so glad I got your and @dduugg's eyes on them.